src/modules/search/components/SearchResults.tsx

```diff
@@ -1,64 +1,86 @@
 import * as React from 'react';
 import {VelocityTransitionGroup} from 'velocity-react';
 import styled from 'styled-components';
-
 import ArrowNavigation from '../../common/components/ArrowNavigation';
 import Track from './Track';
 import {$Result} from './types';
 import zindex from '../../common/zindex';
 import * as animations from '../../common/animations';
+import ResultsFilter from './ResultsFilter';
 
-interface $Props {
-	results: $Result[];
-	enqueue(song: $Result): any;
-}
+export default class SearchResults extends React.Component {
+	props: {
+		results: $Result[];
+		enqueue(song: $Result): any;
+	};
+	state = {
+		filterFor: 'both'
+	};
+
+	handleFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
+		this.setState({
+			filterFor: e.target.id
+		});
+	};
 
-export default function SearchResults({results, enqueue}: $Props) {
-	return (
-		<StyledSearchResults>
-			<ArrowNavigation priority={ArrowNavigation.PRIORITY_MAP.SEARCH} childIsWrapped>
-				<VelocityTransitionGroup
-					enter={{
-						animation: animations.rotate3d.in,
-						stagger: 200,
-						duration: 750,
-						display: 'flex'
-					}}
-					leave={{animation: animations.rotate3d.out, duration: 200, display: 'flex'}}
-					runOnMount
-				>
-					{results.map(result => {
-						switch (result.__typename) {
-							case 'YoutubeResult':
-								return (
-									<Track
-										search
-										thumbnail={result.snippet.thumbnails.default.url}
-										title={result.snippet.title}
-										key={result.id.videoId}
-										onClick={() => enqueue(result)}
-										youtube
-									/>
-								);
-							case 'SoundCloudResult':
-								return (
-									<Track
-										search
-										thumbnail={result.thumbnail}
-										title={result.title}
-										key={result.id}
-										onClick={() => enqueue(result)}
-										soundcloud
-									/>
-								);
-							default:
-								return null;
-						}
-					})}
-				</VelocityTransitionGroup>
-			</ArrowNavigation>
-		</StyledSearchResults>
-	);
+	render() {
+		const {results, enqueue} = this.props;
+		return (
+			<div>
+				<ResultsFilter handleFilter={this.handleFilter} />
+				<StyledSearchResults>
+					<ArrowNavigation priority={ArrowNavigation.PRIORITY_MAP.SEARCH} childIsWrapped>
+						<VelocityTransitionGroup
+							enter={{
+								animation: animations.rotate3d.in,
+								stagger: 200,
+								duration: 750,
+								display: 'flex'
+							}}
+							leave={{animation: animations.rotate3d.out, duration: 200, display: 'flex'}}
+							runOnMount
+						>
+							{results
+								.filter(result => {
+									return this.state.filterFor === 'both' ||
+										result.__typename === this.state.filterFor
+										? result
+										: null;
```

Comment:
This is fine, but just for you own personal knowledge:`filter` doesn't actually use the result of your function. It just converts it to a boolean and uses that to determine if it should include the original value.

So this is identical to 
```javascript
const {filterFor} = this.state
results.filter(result => filterFor === 'both' || result.__typename === filterFor)
```